### DISCUSSION TITLE
chocolatey package is outdated

### DISCUSSION THREAD

**Discussion by RiskManagementGroup** [rabbitmq/rabbitmq-server#6104] [Other]
The chocolatey package is outdated. The last version released to chocolatey is v3.10.6.
See: https://community.chocolatey.org/packages/rabbitmq
As of writing the newest version is v.3.11.0

**Comment by michaelklishin** [comment]
We submit every version but a month or two ago Chocolatey started rejecting them with a single desktop screen for explanation. We are not sure what we can do and what has to change in the package.
If you know how to contact Chocolatey maintainers directly, let us know.

**Reply by pauby** [reply]
If you go to the package page and click 'Contact Maintainers' then it allows you to send a message directly to the maintainers. If that doesn't get a response, then on the same package click 'Contact Site Admins' to send a message to the Site Admins and Moderators.

**Comment by michaelklishin** [comment]
Here is the package source: https://github.com/rabbitmq/chocolatey-package.

**Reply by garrmark** [reply]
I pulled this and ran it locally on a windows 11 machine, the silent install isn't completely silent, it popped this error for me
![image](https://user-images.githubusercontent.com/2669864/197597796-cbc11cd3-2c2b-40b8-a0db-b2466ce8d452.png)

I then ran the install exe to get the error on installing the service, it shows this:
Installing RabbitMQ service...
C:\Program Files\Erlang OTP\erts-13.0.4\bin\erlsrv: Service RabbitMQ added to system.
C:\Program Files\Erlang OTP\erts-13.0.4\bin\erlsrv: Unrecognized option neighbours:.

That's about as far as I got

**Reply by garrmark** [reply]
so the installer errored out, but left the files so here's the output of trying to install the service:

.\rabbitmq-service.bat install

**********************************************
ERLANG_SERVICE_MANAGER_PATH not set correctly.
**********************************************

"\erlsrv.exe" not found
Please set ERLANG_SERVICE_MANAGER_PATH to the folder containing "erlsrv.exe".

**Reply by garrmark** [reply]
my issue was respolved by setting
`$env:HOMEDRIVE=C:`

**Reply by garrmark** [reply]
I submitted https://github.com/rabbitmq/chocolatey-package/pull/59

I don't know if it'll fix the chocolatey install test, but if you could try resubmitting to chocolatey for v3.10.10, that would be cool @michaelklishin

**Comment by michaelklishin** [comment]
I have submitted a [`3.9.23` version](https://gist.github.com/choco-bot/f4d9a79aa74e0793d0f18183690cd3e2) and now we do see a specific error message: it can be that the Erlang dependency in Chocolatey package metadata is out of date.

Unfortunately previously we had a blank (as in, just a desktop) screen without any details. 3.10 and 3.11 might still run into that, we will see possibly even later today.

**Comment by lukebakken** [comment]


**Reply by garrmark** [reply]
when you do a local install and set the source to your current directory, I believe it only looks there for dependencies.

https://docs.chocolatey.org/en-us/create/create-packages#testing-your-package

You'll need to append the API path like so: --source "'.;https://community.chocolatey.org/api/v2/'" (note the double quotes bookending the apostrophes here, use %cd% in cmd.exe or $pwd in Powershell.exe if . doesn't resolve).

**Reply by michaelklishin** [reply]
But there is a [25.0.4 release of the `erlang` package](https://community.chocolatey.org/packages/erlang) according to chcolatey.org :/

**Comment by michaelklishin** [comment]
A few potentially relevant installer issues addressed recently:

 * https://github.com/rabbitmq/rabbitmq-packaging/pull/22
 * https://github.com/rabbitmq/rabbitmq-packaging/pull/23
 * https://github.com/rabbitmq/rabbitmq-packaging/pull/29

**Comment by henebb** [comment]
Any updates on the chocolatey package? The latest version still seems to be 3.10.6.

**Reply by lukebakken** [reply]
We've been having issues for a while now with chocolatey 🤷‍♂️ 

<img width="823" alt="image" src="https://user-images.githubusercontent.com/514926/207642047-0ae74132-a393-4e1b-bffe-d4701347f8bd.png">

**Reply by lukebakken** [reply]
For instance, I have no idea why verification of 3.11.3 failed -  https://community.chocolatey.org/packages/rabbitmq/3.11.3

**Reply by pauby** [reply]
@lukebakken What did you check to troubleshoot this and find out?

Did you see [my comment](https://github.com/rabbitmq/rabbitmq-server/discussions/6104#discussioncomment-3991034) above?

**Reply by lukebakken** [reply]
> What did you check to troubleshoot this and find out?

The link I provided gives no useful information - https://community.chocolatey.org/packages/rabbitmq/3.11.3

Am I missing something in the verification output?

I saw your comment. The RabbitMQ team (which I am a member of) are the maintainers of this package. I suppose it's about time to contact the Chocolatey admins.

**Reply by pauby** [reply]
If you go to the message in the moderation comments on that page it gives you a [link to the logs in the gist](https://gist.github.com/41316b7a9eeddf16c7f157a593e10ca4). That message is also sent by email to all of the maintainers.

![image](https://user-images.githubusercontent.com/12760779/207673660-a13b36c3-0bb6-4a73-a7d7-b39778dc7cfd.png)

The [last message in that log](https://gist.github.com/choco-bot/41316b7a9eeddf16c7f157a593e10ca4#file-install-txt-L741) says:

```
Elevating permissions and running ["C:\Users\Administrator\AppData\Local\Temp\chocolatey\rabbitmq\3.11.3\rabbitmq-server-3.11.3.exe" /S ]. This may take a while, depending on the statements.
```

Which means that the install timed out. The screenshot at the bottom of the gist also shows a black window. This generally means that the installer is not silent and waiting on input.

Have you tested the package in the [Chocolatey Test Environment](https://github.com/chocolatey-community/chocolatey-test-environment/)? You may find the software needs a dependency (maybe a version of .NET as an example) that is not present on the system and should be added as a package dependency.

**Reply by lukebakken** [reply]
Yes, I followed that link and noticed that the log didn't say anything useful. It would be great if there were an actual "installer timed out" message somewhere.

> Which means that the install timed out

...and yet the RabbitMQ team knows that the Windows installer works just fine, because people who use Windows and *pay for RabbitMQ support* would have raised hell about it by now.

The only dependency RabbitMQ needs is Erlang. This has not changed since the dawn of time.

I'll try to find time to rectify this situation but as you can imagine, there are higher priorities. Nobody who pays for RabbitMQ support has requested this fix. That would raise its priority.

**Reply by michaelklishin** [reply]
Yes, recently there were improvements around debugging of Chocolatey package submissions. It's just hasn't been a priority compared to other things.

**Reply by lukebakken** [reply]
@pauby I appreciate you pointing out the chocolatey test environment and I'll give that a try.

**Reply by lukebakken** [reply]
Installing 3.11.5 worked just fine using the test environment:

<img width="983" alt="image" src="https://user-images.githubusercontent.com/514926/207721134-760a889c-72e4-4496-be7b-8e650003ac35.png">

**Comment by lukebakken** [comment]
@pauby @michaelklishin @henebb @garrmark @RiskManagementGroup 

RabbitMQ 3.11.5 has been approved. I have no idea why other versions did not work:

https://community.chocolatey.org/packages/rabbitmq

**Reply by henebb** [reply]
Nice work! 👏

**Comment by pauby** [comment]
@lukebakken 

> ...and yet the RabbitMQ team knows that the Windows installer works just fine, because people who use Windows and pay for RabbitMQ support would have raised hell about it by now.

> I'll try to find time to rectify this situation but as you can imagine, there are higher priorities. Nobody who pays for RabbitMQ support has requested this fix. That would raise its priority.

Just to be very clear, I jumped in here to help out. I know it's difficult and inadvisable to read tone into messages, but the tone of yours was frustration with me.

> RabbitMQ 3.11.5 has been approved. I have no idea why other versions did not work:

I'm going to take a guess. 

We recently (as in the last few days) upgraded our test environment to use Windows Server 2019 as a base. We are still working on tweaking this so, we have not updated the test environment repository I pointed you to. That will likely be updated in the new year when a blog post will also follow.

Having said that, it doesn't make a lot of sense when you tried it in the 2012 R2 test environment and it worked. Because that is the same version that the previous versions used. So, I have to throw my hands up and say I don't know.

It's working now and that is all that matters.

If you need to have those other versions that were rejected, available, please let me know.

**Reply by lukebakken** [reply]
> I know it's difficult and inadvisable to read tone into messages, but the tone of yours was frustration with me

I'm appreciative of your assistance, thank you.

I'm frustrated with _other users_ who have repeatedly commented "where's the latest version" without doing anything to assist. You know how it is with free software.

I can't explain why it worked using Windows 2012 in my environment. The only thing I could say is that I logged into the running VM and ran `choco install...` there, and didn't use the provisioning script. That could be where the difference lies.

I don't think we need to worry about the other versions at this time.

Thanks again for your assistance and Merry Christmas / Happy New Year!

**Comment by pauby** [comment]
> I'm frustrated with other users who have repeatedly commented "where's the latest version" without doing anything to assist. You know how it is with free software.

😂 I was actually thinking the very same thing not less than 30 minutes ago. Unfortunately, I completely appreciate that feeling.

> Thanks again for your assistance and Merry Christmas / Happy New Year!

To you, and the whole team! 

We appreciate you maintaining the package and persevering through the issues. If you have problems in the future, please let me know.

**Reply by lukebakken** [reply]
Thanks again! Our conversation reminded me that it's time to renew my chocolatey pro license.
